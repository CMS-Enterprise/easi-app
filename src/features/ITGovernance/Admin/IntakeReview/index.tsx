import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  ITGovIntakeFormStatus,
  SystemIntakeFragmentFragment
} from 'gql/generated/graphql';

import PageHeading from 'components/PageHeading';
import PDFExport from 'components/PDFExport';
import SystemIntakeReview from 'components/SystemIntakeReview';
import TaskStatusTag from 'components/TaskStatusTag';
import { formatDateLocal } from 'utils/date';

type IntakeReviewProps = {
  systemIntake: SystemIntakeFragmentFragment;
};

const IntakeReview = ({ systemIntake }: IntakeReviewProps) => {
  const { t } = useTranslation('governanceReviewTeam');
  const filename = `System intake for ${systemIntake.requestName}.pdf`;

  const intakeStatus = systemIntake.itGovTaskStatuses.intakeFormStatus;
  let dateInfo: string | undefined;

  if (
    intakeStatus === ITGovIntakeFormStatus.EDITS_REQUESTED &&
    systemIntake.updatedAt
  ) {
    dateInfo = systemIntake.updatedAt;
  } else if (
    intakeStatus === ITGovIntakeFormStatus.COMPLETED &&
    systemIntake.submittedAt
  ) {
    dateInfo = systemIntake.submittedAt;
  }

  // this is the admin view

  return (
    <div data-testid="intake-review">
      <div className="margin-bottom-4">
        <PageHeading className="margin-y-0">{t('general:intake')}</PageHeading>
        {!!intakeStatus && (
          <>
            <TaskStatusTag status={intakeStatus} />
            <span className="text-base">
              {t('general:on')}
              {formatDateLocal(dateInfo, 'MM/dd/yyyy')}
            </span>
          </>
        )}
      </div>
      <PDFExport
        title="System Intake"
        filename={filename}
        label={t('intake:viewIntakeRequest.downloadPDF')}
        linkPosition="both"
        helpText={t('intake:viewIntakeRequest.docsNotIncluded')}
      >
        <div className="border-top border-base-light margin-top-2">
          <SystemIntakeReview systemIntake={systemIntake} />
        </div>
      </PDFExport>
    </div>
  );
};

export default IntakeReview;
