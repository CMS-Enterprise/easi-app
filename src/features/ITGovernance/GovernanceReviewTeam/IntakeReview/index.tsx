import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { SystemIntake } from 'gql/legacyGQL/types/SystemIntake';

import UswdsReactLink from 'components/LinkWrapper';
import PageHeading from 'components/PageHeading';
import PDFExport from 'components/PDFExport';
import SystemIntakeReview from 'components/SystemIntakeReview';

import ITGovAdminContext from '../ITGovAdminContext';

type IntakeReviewProps = {
  systemIntake: SystemIntake;
};

const IntakeReview = ({ systemIntake }: IntakeReviewProps) => {
  const { t } = useTranslation('governanceReviewTeam');
  const filename = `System intake for ${systemIntake.requestName}.pdf`;

  const isITGovAdmin = useContext(ITGovAdminContext);

  return (
    <div data-testid="intake-review">
      <PageHeading className="margin-top-0">{t('general:intake')}</PageHeading>
      <PDFExport
        title="System Intake"
        filename={filename}
        label="Download System Intake as PDF"
      >
        <SystemIntakeReview systemIntake={systemIntake} />
      </PDFExport>

      {isITGovAdmin && (
        <UswdsReactLink
          className="usa-button margin-top-5"
          variant="unstyled"
          to={`/it-governance/${systemIntake.id}/actions`}
        >
          {t('action:takeAnAction')}
        </UswdsReactLink>
      )}
    </div>
  );
};

export default IntakeReview;
