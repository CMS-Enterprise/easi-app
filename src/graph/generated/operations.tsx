import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';

const defaultOptions = {};
export type GetSystemIntakeQueryVariables = Exact<{
  id: Scalars['UUID'];
}>;

export type GetSystemIntakeQuery = { __typename?: 'Query' } & {
  systemIntake?: Maybe<
    { __typename?: 'SystemIntake' } & Pick<
      SystemIntake,
      | 'id'
      | 'adminLead'
      | 'businessNeed'
      | 'businessSolution'
      | 'currentStage'
      | 'grbDate'
      | 'grtDate'
      | 'lcid'
      | 'needsEaSupport'
      | 'requestName'
      | 'requestType'
      | 'status'
      | 'submittedAt'
    > & {
        businessOwner?: Maybe<
          { __typename?: 'SystemIntakeBusinessOwner' } & Pick<
            SystemIntakeBusinessOwner,
            'component' | 'name'
          >
        >;
        contract?: Maybe<
          { __typename?: 'SystemIntakeContract' } & Pick<
            SystemIntakeContract,
            'contractor' | 'hasContract' | 'vehicle'
          > & {
              endDate?: Maybe<
                { __typename?: 'ContractDate' } & Pick<
                  ContractDate,
                  'day' | 'month' | 'year'
                >
              >;
              startDate?: Maybe<
                { __typename?: 'ContractDate' } & Pick<
                  ContractDate,
                  'day' | 'month' | 'year'
                >
              >;
            }
        >;
        costs?: Maybe<
          { __typename?: 'SystemIntakeCosts' } & Pick<
            SystemIntakeCosts,
            'isExpectingIncrease' | 'expectedIncreaseAmount'
          >
        >;
        grtFeedbacks: Array<
          { __typename?: 'GRTFeedback' } & Pick<
            GrtFeedback,
            'feedback' | 'feedbackType' | 'createdAt'
          >
        >;
        governanceTeams?: Maybe<
          { __typename?: 'SystemIntakeGovernanceTeam' } & Pick<
            SystemIntakeGovernanceTeam,
            'isPresent'
          > & {
              teams?: Maybe<
                Array<
                  { __typename?: 'SystemIntakeCollaborator' } & Pick<
                    SystemIntakeCollaborator,
                    'acronym' | 'collaborator' | 'key' | 'label' | 'name'
                  >
                >
              >;
            }
        >;
        isso?: Maybe<
          { __typename?: 'SystemIntakeISSO' } & Pick<
            SystemIntakeIsso,
            'isPresent' | 'name'
          >
        >;
        fundingSource?: Maybe<
          { __typename?: 'SystemIntakeFundingSource' } & Pick<
            SystemIntakeFundingSource,
            'fundingNumber' | 'isFunded' | 'source'
          >
        >;
        productManager?: Maybe<
          { __typename?: 'SystemIntakeProductManager' } & Pick<
            SystemIntakeProductManager,
            'component' | 'name'
          >
        >;
        requester: { __typename?: 'SystemIntakeRequester' } & Pick<
          SystemIntakeRequester,
          'component' | 'email' | 'name'
        >;
      }
  >;
};

export const GetSystemIntakeDocument = gql`
  query GetSystemIntake($id: UUID!) {
    systemIntake(id: $id) {
      id
      adminLead
      businessNeed
      businessSolution
      businessOwner {
        component
        name
      }
      contract {
        contractor
        endDate {
          day
          month
          year
        }
        hasContract
        startDate {
          day
          month
          year
        }
        vehicle
      }
      costs {
        isExpectingIncrease
        expectedIncreaseAmount
      }
      currentStage
      grbDate
      grtDate
      grtFeedbacks {
        feedback
        feedbackType
        createdAt
      }
      governanceTeams {
        isPresent
        teams {
          acronym
          collaborator
          key
          label
          name
        }
      }
      isso {
        isPresent
        name
      }
      fundingSource {
        fundingNumber
        isFunded
        source
      }
      lcid
      needsEaSupport
      productManager {
        component
        name
      }
      requester {
        component
        email
        name
      }
      requestName
      requestType
      status
      submittedAt
    }
  }
`;

/**
 * __useGetSystemIntakeQuery__
 *
 * To run a query within a React component, call `useGetSystemIntakeQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetSystemIntakeQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetSystemIntakeQuery({
 *   variables: {
 *      id: // value for 'id'
 *   },
 * });
 */
export function useGetSystemIntakeQuery(
  baseOptions: Apollo.QueryHookOptions<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useQuery<GetSystemIntakeQuery, GetSystemIntakeQueryVariables>(
    GetSystemIntakeDocument,
    options
  );
}
export function useGetSystemIntakeLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useLazyQuery<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >(GetSystemIntakeDocument, options);
}
export type GetSystemIntakeQueryHookResult = ReturnType<
  typeof useGetSystemIntakeQuery
>;
export type GetSystemIntakeLazyQueryHookResult = ReturnType<
  typeof useGetSystemIntakeLazyQuery
>;
export type GetSystemIntakeQueryResult = Apollo.QueryResult<
  GetSystemIntakeQuery,
  GetSystemIntakeQueryVariables
>;
