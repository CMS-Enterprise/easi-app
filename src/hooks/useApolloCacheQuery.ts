// useApolloCacheQuery.ts
import {
  DocumentNode,
  OperationVariables,
  QueryHookOptions,
  QueryResult,
  TypedDocumentNode,
  useApolloClient,
  useQuery
} from '@apollo/client';

/** Custom hook to query the cache to determine if query result exists already */
export default function useApolloCacheQuery<
  TData = any,
  TVariables = OperationVariables
>(
  query: DocumentNode | TypedDocumentNode<TData, TVariables>,
  options?: QueryHookOptions<TData, TVariables>
): QueryResult<TData, TVariables> {
  const { cache } = useApolloClient();

  const queryResult = useQuery<TData, TVariables>(query, options);

  return {
    ...queryResult,
    loading:
      !cache.diff({
        query: cache.transformDocument(query),
        variables: options?.variables,
        returnPartialData: true,
        optimistic: false
      }).complete && queryResult.loading
  };
}
