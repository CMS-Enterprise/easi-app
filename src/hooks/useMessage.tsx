import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
