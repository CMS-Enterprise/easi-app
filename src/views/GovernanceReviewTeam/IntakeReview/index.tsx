import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { Link as UswdsLink } from '@trussworks/react-uswds';
import { DateTime } from 'luxon';

import SystemIntakeReview from 'components/SystemIntakeReview';
import { SystemIntakeForm } from 'types/systemIntake';

type IntakeReviewProps = {
  systemIntake: SystemIntakeForm;
  now: DateTime;
};

const IntakeReview = ({ systemIntake, now }: IntakeReviewProps) => {
  const { t } = useTranslation('governanceReviewTeam');

  return (
    <div>
      <h1 className="margin-top-0">{t('general:intake')}</h1>
      <SystemIntakeReview systemIntake={systemIntake} now={now} />
      <UswdsLink
        className="usa-button margin-top-5"
        variant="unstyled"
        to={`/governance-review-team/${systemIntake.id}/actions`}
        asCustom={Link}
      >
        Take an action
      </UswdsLink>
    </div>
  );
};

export default IntakeReview;
