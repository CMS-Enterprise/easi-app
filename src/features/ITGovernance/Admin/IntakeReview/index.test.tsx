import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen } from '@testing-library/react';
import { SystemIntakeFragmentFragment } from 'gql/generated/graphql';
import {
  getSystemIntakeContactsQuery,
  getSystemIntakeQuery,
  requester,
  systemIntake
} from 'tests/mock/systemIntake';

import { MessageProvider } from 'hooks/useMessage';
import VerboseMockedProvider from 'utils/testing/VerboseMockedProvider';

import IntakeReview from './index';

describe('The GRT intake review view', () => {
  let dateSpy: any;
  beforeAll(() => {
    // September 30, 2020
    dateSpy = vi.spyOn(Date, 'now').mockImplementation(() => 1601449200000);
  });

  afterAll(() => {
    dateSpy.mockRestore();
  });

  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <MockedProvider
          mocks={[getSystemIntakeQuery(), getSystemIntakeContactsQuery]}
        >
          <MessageProvider>
            <IntakeReview systemIntake={systemIntake} />
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(screen.getByTestId('intake-review')).toBeInTheDocument();
  });

  it('matches the snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[`/it-governance/${systemIntake.id}/intake-request`]}
      >
        <MockedProvider
          mocks={[getSystemIntakeQuery(), getSystemIntakeContactsQuery]}
        >
          <Route path={['/it-governance/:systemId/intake-request']}>
            <MessageProvider>
              <IntakeReview systemIntake={systemIntake} />
            </MessageProvider>
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(
      await screen.findByTestId(`contact-requester-${requester.euaUserId}`)
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders increased costs data', () => {
    const costs: SystemIntakeFragmentFragment['costs'] = {
      __typename: 'SystemIntakeCosts',
      isExpectingIncrease: 'YES',
      expectedIncreaseAmount: 'less than $1 million'
    };

    render(
      <MemoryRouter>
        <VerboseMockedProvider
          mocks={[
            getSystemIntakeQuery({ costs }),
            getSystemIntakeContactsQuery
          ]}
        >
          <MessageProvider>
            <IntakeReview
              systemIntake={{
                ...systemIntake,
                costs
              }}
            />
          </MessageProvider>
        </VerboseMockedProvider>
      </MemoryRouter>
    );

    expect(screen.getByText(/less than \$1 million/i)).toBeInTheDocument();
  });

  it('renders annual spending data', () => {
    const annualSpending: SystemIntakeFragmentFragment['annualSpending'] = {
      __typename: 'SystemIntakeAnnualSpending',
      currentAnnualSpending: '3.50',
      currentAnnualSpendingITPortion: '35',
      plannedYearOneSpending: '123456',
      plannedYearOneSpendingITPortion: '50'
    };

    render(
      <MemoryRouter>
        <MockedProvider
          mocks={[
            getSystemIntakeQuery({ annualSpending }),
            getSystemIntakeContactsQuery
          ]}
        >
          <MessageProvider>
            <IntakeReview
              systemIntake={{
                ...systemIntake,
                annualSpending
              }}
            />
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(screen.getByText('$3.5')).toBeInTheDocument();
    expect(screen.getByText('$123,456')).toBeInTheDocument();
  });
});
