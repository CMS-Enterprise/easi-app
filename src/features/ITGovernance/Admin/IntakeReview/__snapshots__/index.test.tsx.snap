// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The GRT intake review view > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="intake-review"
  >
    <h1
      aria-live="polite"
      class="easi-h1 margin-top-0"
      tabindex="-1"
    >
      Intake Request
    </h1>
    <div
      class="easi-pdf-export"
    >
      <h1
        class="easi-only-print"
      >
        System Intake
      </h1>
      <div>
        <dl
          title="System Request"
        >
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Submission date
              </dt>
              <dd
                class="description-definition margin-0"
              >
                October 20, 2022
              </dd>
            </div>
          </div>
        </dl>
        <hr
          class="system-intake__hr"
        />
        <h2
          class="font-heading-xl"
        >
          Contact details
        </h2>
        <dl
          title="Contact details"
        >
          <div
            class="easi-review-row"
          >
            <div
              data-testid="contact-requester-A11Y"
            >
              <dt
                class="text-bold margin-bottom-05"
              >
                Requester
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Ally Anderson
              </dd>
            </div>
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Requester component
              </dt>
              <dd
                class="description-definition margin-0"
              >
                CMS Wide
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                CMS Business Owner's name
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Adeline Aarons
              </dd>
            </div>
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                CMS Business Owner component
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Center for Medicare
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                CMS Project/Product Manager or Lead
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Audrey Abrams
              </dd>
            </div>
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                CMS Project/Product Manager or Lead component
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Office of Legislation
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Does your project have an Information System Security Officer (ISSO)?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Yes, Aaron Adams
              </dd>
            </div>
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                I have started collaborating with
              </dt>
              <dd
                class="description-definition margin-0"
              >
                N/A
              </dd>
            </div>
          </div>
        </dl>
        <hr
          class="system-intake__hr"
        />
        <h2
          class="font-heading-xl"
        >
          Request details
        </h2>
        <dl
          title="Request details"
        >
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Project name
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Mock System Intake Request
              </dd>
            </div>
          </div>
          <div
            class="margin-bottom-205 line-height-body-3"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                What is your business need?
              </dt>
              <dd
                class="description-definition margin-0 text-pre-wrap"
              >
                The quick brown fox jumps over the lazy dog.
              </dd>
            </div>
          </div>
          <div
            class="margin-bottom-205 line-height-body-3"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                How are you thinking of solving it?
              </dt>
              <dd
                class="description-definition margin-0 text-pre-wrap"
              >
                The quick brown fox jumps over the lazy dog.
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Where are you in the process?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                The quick brown fox jumps over the lazy dog.
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Do you need Enterprise Architecture (EA) support?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                No
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Does your request involve AI technologies?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                Yes
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Does your project involve any user interface component, or changes to an interface component?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                No
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Do you plan to use any software products to fulfill your business needs?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                No
              </dd>
            </div>
          </div>
        </dl>
        <hr
          class="system-intake__hr"
        />
        <h2
          class="font-heading-xl"
        >
          Contract details
        </h2>
        <dl
          title="Contract details"
        >
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Which existing funding sources will fund this project?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                N/A
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Do the costs for this request exceed what you are currently spending to meet your business need?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                No
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Do you already have a contract in place to support this effort?
              </dt>
              <dd
                class="description-definition margin-0"
              >
                I am currently working on my OAGM Acquisition Plan/IAA documents
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Contractors
              </dt>
              <dd
                class="description-definition margin-0"
              >
                TrussWorks, Inc.
              </dd>
            </div>
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Contract number
              </dt>
              <dd
                class="description-definition margin-0"
              >
                123456-7890
              </dd>
            </div>
          </div>
          <div
            class="easi-review-row"
          >
            <div>
              <dt
                class="text-bold margin-bottom-05"
              >
                Period of performance
              </dt>
              <dd
                class="description-definition margin-0"
              >
                1/2020 to 12/2020
              </dd>
            </div>
          </div>
        </dl>
        <hr
          class="system-intake__hr"
        />
        <h2
          class="font-heading-xl"
        >
          Documents
        </h2>
        <div
          id="systemIntakeDocuments"
        >
          <div
            class="usa-table-container--scrollable"
            data-testid="scrollable-table-container"
            tabindex="0"
          >
            <table
              class="usa-table usa-table--borderless width-full"
              data-testid="table"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      File name
                      <svg
                        aria-label="unfold more"
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Document type
                      <svg
                        aria-label="unfold more"
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="descending"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Date added
                      <svg
                        aria-label="expand"
                        class="usa-icon margin-left-05"
                        data-testid="caret--down"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Actions
                      <svg
                        aria-label="unfold more"
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              />
            </table>
          </div>
          <p>
            No documents uploaded
          </p>
        </div>
      </div>
    </div>
    <div
      class="easi-pdf-export__controls margin-top-6"
    >
      <button
        class="usa-button usa-button--unstyled display-flex flex-align-center easi-no-print"
        data-testid="button"
        type="button"
      >
        <svg
          aria-label="download"
          class="usa-icon margin-right-1"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
        Download System Intake as PDF
      </button>
    </div>
  </div>
</DocumentFragment>
`;
